apis/v1/gateway_types.go

```diff
@@ -1237,6 +1247,50 @@ const (
 	GatewayReasonNotReconciled GatewayConditionReason = "NotReconciled"
 )
 
+const (
+	// This condition indicates whether the controller was able to resolve all
+	// the object references for the Gateway that are not part of a specific 
+	// Listener configuration.
+	//
+	// Possible reasons for this condition to be True are:
+	//
+	// * "ResolvedRefs"
+	//
+	// Possible reasons for this condition to be False are:
+	//
+	// * "RefNotPermitted"
+	// * "InvalidClientCertificateRef"
```

Comment:
Maybe it would also make sense to add a reason (e.g., `ListenersNotResolved`) to indicate when there are unresolved Listeners. This would allow the condition to serve as a summary of all Listener-level ResolvedRefs conditions.

Reply:
Yes, I think that is fine.